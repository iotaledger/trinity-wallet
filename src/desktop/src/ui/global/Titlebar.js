/*global Electron*/
import React from 'react';
import css from './titlebar.scss';

/**
 * Button component
 */
const Titlebar = () => {
    const os = Electron.getOS();

    const windows = () => {
        return (
            <nav className={css.windows}>
                <a onClick={() => Electron.showMenu()}>
                    <svg width="15" height="15" viewBox="0 0 15 15">
                        <path fill="#000" d="M0 1h15v1h-15z" />
                        <path fill="#000" d="M0 7h15v1h-15z" />
                        <path fill="#000" d="M0 13h15v1h-15z" />
                    </svg>
                </a>
                <a onClick={() => Electron.minimize()}>
                    <svg width="15" height="15" viewBox="0 0 15 15">
                        <path fill="#000" d="M0 12h15v1h-15z" />
                    </svg>
                </a>
                <a onClick={() => Electron.maximize()}>
                    <svg width="15" height="15" viewBox="0 0 15 15">
                        <path d="M1 1v13h13v-13h-13zm-1-1h15v15h-15v-15z" />
                    </svg>
                </a>
                <a onClick={() => Electron.close()}>
                    <svg width="15" height="15" viewBox="0 0 15 15">
                        <path d="M7.425 6.718l6.718-6.718.707.707-6.718 6.718 6.718 6.718-.707.707-6.718-6.718-6.718 6.718-.707-.707 6.718-6.718-6.718-6.718.707-.707 6.718 6.718z" />
                    </svg>
                </a>
            </nav>
        );
    };

    return <div className={css.titlebar}>{os === 'win32' ? windows() : null}</div>;
};

export default Titlebar;
