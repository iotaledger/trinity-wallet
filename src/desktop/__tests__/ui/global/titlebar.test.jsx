import React from 'react';
import { shallow } from 'enzyme';

import Titlebar from 'ui/global/Titlebar';

const props = {
    path: '/',
};

global.Electron = {
    getOS: () => 'win32',
};

describe('Titlebar component', () => {
    test('Render the component', () => {
        const wrapper = shallow(<Titlebar {...props} />);

        expect(wrapper).toMatchSnapshot();
    });

    test('Render dark titlebar', () => {
        const wrapper = shallow(<Titlebar path="settings" />);

        expect(wrapper.find('.dark')).toHaveLength(1);
    });

    test('Render light titlebar', () => {
        const wrapper = shallow(<Titlebar path="onboarding" />);

        expect(wrapper.find('.light')).toHaveLength(1);
    });
});
