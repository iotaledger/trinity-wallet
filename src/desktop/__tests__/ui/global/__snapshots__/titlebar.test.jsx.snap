// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Titlebar component Render the component 1`] = `
<div
  className="titlebar"
>
  <nav
    className="windows"
  >
    <a
      className="light"
      onClick={[Function]}
    >
      <svg
        height="15"
        viewBox="0 0 15 15"
        width="15"
      >
        <path
          d="M0 1h15v1h-15z"
          fill="#000"
        />
        <path
          d="M0 7h15v1h-15z"
          fill="#000"
        />
        <path
          d="M0 13h15v1h-15z"
          fill="#000"
        />
      </svg>
    </a>
    <a
      onClick={[Function]}
    >
      <svg
        height="15"
        viewBox="0 0 15 15"
        width="15"
      >
        <path
          d="M0 12h15v1h-15z"
          fill="#000"
        />
      </svg>
    </a>
    <a
      onClick={[Function]}
    >
      <svg
        height="15"
        viewBox="0 0 15 15"
        width="15"
      >
        <path
          d="M1 1v13h13v-13h-13zm-1-1h15v15h-15v-15z"
        />
      </svg>
    </a>
    <a
      onClick={[Function]}
    >
      <svg
        height="15"
        viewBox="0 0 15 15"
        width="15"
      >
        <path
          d="M7.425 6.718l6.718-6.718.707.707-6.718 6.718 6.718 6.718-.707.707-6.718-6.718-6.718 6.718-.707-.707 6.718-6.718-6.718-6.718.707-.707 6.718 6.718z"
        />
      </svg>
    </a>
  </nav>
</div>
`;
